Source/Core/Core/PowerPC/JitLLVM/Jit.cpp

```diff
@@ -0,0 +1,295 @@
+#include <llvm/Transforms/IPO/PassManagerBuilder.h>
+
+#include "Common/GekkoDisassembler.h"
+
+#include "Core/PowerPC/CPUCoreBase.h"
+#include "Core/PowerPC/PPCAnalyst.h"
+
+#include "Core/PowerPC/JitLLVM/Jit.h"
+#include "Core/PowerPC/JitLLVM/JitFunction.h"
+#include "Core/PowerPC/JitLLVM/JitHelpers.h"
+#include "Core/PowerPC/JitLLVM/JitLLVM_Tables.h"
+
+using namespace llvm;
+
+void JitLLVM::Init()
+{
+	JitLLVMTables::InitTables();
+
+	m_debug_enabled = false;
+
+	code_block.m_stats = &js.st;
+	code_block.m_gpa = &js.gpa;
+	code_block.m_fpa = &js.fpa;
+
+	analyzer.SetOption(PPCAnalyst::PPCAnalyzer::OPTION_CONDITIONAL_CONTINUE);
+
+	InitializeNativeTarget();
+	InitializeNativeTargetAsmPrinter();
+	LLVMContext& con = getGlobalContext();
+
+	m_named_binder = new LLVMNamedBinding();
+	m_mem_manager = new LLVMMemoryManager(m_named_binder);
+	m_main_mod = new Module("Dolphin LLVM JIT Recompiler", con);
+
+	m_engine = EngineBuilder(std::unique_ptr<Module>(m_main_mod))
+		.setEngineKind(EngineKind::JIT)
+		.setMCJITMemoryManager(std::unique_ptr<RTDyldMemoryManager>(m_mem_manager))
+		.create();
+}
+
+void JitLLVM::Shutdown()
+{
+	// The ExecutionEngine takes ownership of nearly everything
+	// Delete the engine, delete the world.
+	delete m_engine;
+	delete m_named_binder;
+}
+
+void JitLLVM::DoDownCount()
+{
+	LLVMFunction* func = GetCurrentFunc();
+	IRBuilder<>* builder = func->GetBuilder();
+
+	Value* downcount = func->LoadDownCount();
+	Value* res = builder->CreateSub(downcount, builder->getInt32(js.downcountAmount));
+	func->StoreDownCount(res);
+}
+
+void JitLLVM::WriteExceptionExit(Value* val)
+{
+	LLVMFunction* func = GetCurrentFunc();
+
+	IRBuilder<>* builder = func->GetBuilder();
+
+	DoDownCount();
+
+	func->StorePC(val);
+	func->StoreNPC(val);
+
+	Function* check_exceptions_func = m_cur_binder->GetBinding((void*)&PowerPC::CheckExceptions);
+	builder->CreateCall(check_exceptions_func);
+
+	func->StorePC(func->LoadNPC());
+
+	Function* dispatcher_func = m_cur_binder->GetBinding(m_cur_binder->GetBinding("Dispatcher"));
+	builder->CreateCall(dispatcher_func);
+	builder->CreateRetVoid();
+}
+
+void JitLLVM::WriteExit(u32 destination)
+{
+	LLVMFunction* func = GetCurrentFunc();
+
+	IRBuilder<>* builder = GetCurrentFunc()->GetBuilder();
+
+	DoDownCount();
+
+	func->StorePC(builder->getInt32(destination));
+
+	Function* dispatcher_func = m_cur_binder->GetBinding(m_cur_binder->GetBinding("Dispatcher"));
+	builder->CreateCall(dispatcher_func);
+	builder->CreateRetVoid();
+}
+
+void JitLLVM::WriteExitInValue(Value* destination)
+{
+	LLVMFunction* func = GetCurrentFunc();
+
+	IRBuilder<>* builder = GetCurrentFunc()->GetBuilder();
+
+	DoDownCount();
+
+	func->StorePC(destination);
+
+	Function* dispatcher_func = m_cur_binder->GetBinding(m_cur_binder->GetBinding("Dispatcher"));
+	builder->CreateCall(dispatcher_func);
+	builder->CreateRetVoid();
+}
+
+void JitLLVM::FallBackToInterpreter(LLVMFunction* func, UGeckoInstruction inst)
```

Comment:
Could you please also implement those two FL_ENDBLOCK checks in https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/PowerPC/JitArm64/Jit.cpp#L58 ? So you'll be able to fallback branching and system instructions.